pkg/forwarder/transactions_file_storage.go

```diff
@@ -33,18 +33,18 @@ type transactionsFileStorage struct {
 func newTransactionsFileStorage(
 	serializer *TransactionsSerializer,
 	storagePath string,
-	maxSizeInBytes int64,
+	maxStorage *forwarderMaxStorage,
```

Comment:
Just a drive-by comment: I think what you might consider for easier testing and a bit more modularity is to split all storage-related helpers into a separate object/package and you tie it all together here. Having a separate holder (e.g. `ForwarderStorage{}` object or `storage` package) with some funcs (e.g. `StoragePath()`, `MaxAllowedStorage()`. `ListRetryFiles() []RetryFile`, `RemoveRetryFile(*RetryFile)`, `Refresh()`) can then reduce the complexity here quite a bit and let you test things with in much higher granularity.

It's not a blocker but maybe something to think about if you keep working on this.

Reply:
💯 

For the initial implementation of `transactions_file_storage.go` I decided to not using interface that abstract the file system for unit testing purpose. It was something I did it a few time in the past but noticed the following drawbacks:
- Most of my issues come from the filesystem API (invalid path, relative path vs full path issue), restriction in the name of the file which are not caught when mocking the filesystem. 
- Implementing the mock functions requires quite lot of code.

Now I tend to not mock the filesystem for testing purpose in the general case  (of course there are cases where it is appropriate).

In term of granularity, I tend to split a file when it grows as it is not always easy to guess how the code will evolve.
I think the `pkg/forwarder/transactions_file_storage.go`  is quite easy to understand and small (less than 210 lines).
If I need to add more code (let's say 100 lines or more) I will see what is the most logical way to split the file. What you suggested would be good candidate.